py/picca/delta_extraction/quasar_catalogues/desi_quasar_catalogue.py

```diff
@@ -126,29 +126,49 @@ def add_healpix(self):
         self.catalogue.sort("HEALPIX")
 
     def read_catalogue(self):
-        """Read the DESI quasar catalogue"""
+        """Read the DESI quasar catalogue
+
+        Raise
+        -----
+        QuasarCatalogueError if the catalogue has missing columns or is
+        empty after the filters are applied
+        """
         self.logger.progress(f'Reading catalogue from {self.filename}')
         extnames = [ext.get_extname() for ext in fitsio.FITS(self.filename)]
         if "QSO_CAT" in extnames:
             extension = "QSO_CAT"
         elif "ZCATALOG" in extnames:
             extension = "ZCATALOG"
         else:
-            raise QuasarCatalogueError(
-                f"Could not find valid quasar catalog extension in fits file: {self.filename}")
+            # TODO: this is a patch that should be removed before merging with master
+            # The extension=1 line should be removed and the raise uncommented
+            extension = 1
+            #raise QuasarCatalogueError(
+            #    f"Could not find valid quasar catalog extension in fits file: {self.filename}")
         catalogue = Table(fitsio.read(self.filename, ext=extension))
 
         if 'TARGET_RA' in catalogue.colnames:
             catalogue.rename_column('TARGET_RA', 'RA')
             catalogue.rename_column('TARGET_DEC', 'DEC')
 
+        # mandatory columns
         keep_columns = ['RA', 'DEC', 'Z', 'TARGETID']
+        for col in keep_columns:
+            if col not in catalogue.colnames:
+                raise QuasarCatalogueError(
+                    f"Missing required column {col} in quasar catalogue")
+
+        # optional columns
         if 'TILEID' in catalogue.colnames:
             keep_columns += ['TILEID', 'PETAL_LOC']
+            if 'PETAL_LOC' not in catalogue.colnames:
+                raise QuasarCatalogueError(
+                    "When TILEID is in the catalogue, PETAL_LOC is also "
+                    "expected to be present but it is not.")
```

Comment:
When does this happen? No harm to have the check, but I'd be surprise that this ever happens.

Reply:
I'm not sure if this will ever happen. When I changed LAST_NIGHT to LASTNIGHT I saw that we were not checking that the selected columns were actually present so I added the checks